lhotse/recipes/voxceleb.py

```diff
@@ -0,0 +1,333 @@
+
+    NOTE: We prepare the data using the Kaldi style split, i.e., the whole VoxCeleb2
+    ("dev" and "test") and the training portion ("dev") of VoxCeleb1 are put into the
+    "train" split. The "test" split contains the "test" portion of VoxCeleb1. So if
+    VoxCeleb1 is not provided, no "test" split is created in the output manifests.
+
+    Example usage:
+    >>> from lhotse.recipes.voxceleb import prepare_voxceleb
+    >>> manifests = prepare_voxceleb(voxceleb_v1_root='/path/to/voxceleb1',
+    ...                               voxceleb_v2_root='/path/to/voxceleb2',
+    ...                               output_dir='/path/to/output',
+    ...                               num_jobs=4)
+
+    NOTE: If VoxCeleb1 is provided, we also prepare the trials file using the list provided
+    in http://www.openslr.org/resources/49/voxceleb1_test_v2.txt. This file is used in the
+    Kaldi recipes for VoxCeleb speaker verification. This is prepared as 2 tuples of the form
+    (CutSet, CutSet) with identical id's, one for each of positive pairs and negative pairs.
+    These are stored in the dict under keys 'pos_trials' and 'neg_trials', respectively.
+    For evaluation purpose, the :func:`<lhotse.dataset.sampling.CutPairsSampler>`
+    can be used to sample from this tuple.
+    """
+    voxceleb1_root = Path(voxceleb1_root) if voxceleb1_root else None
+    voxceleb2_root = Path(voxceleb2_root) if voxceleb2_root else None
+    if not (voxceleb1_root or voxceleb2_root):
+        raise ValueError("Either VoxCeleb1 or VoxCeleb2 path must be provided.")
+
+    output_dir = Path(output_dir) if output_dir is not None else None
+    manifests = defaultdict(dict)
+    if voxceleb1_root:
+        logging.info("Preparing VoxCeleb1...")
+        manifests.update(_prepare_voxceleb_v1(voxceleb1_root, num_jobs))
+        manifests.update(_prepare_voxceleb_trials(manifests["test"]))
+    else:
+        logging.info(
+            "VoxCeleb1 not provided, no test split or trials file will be created..."
+        )
+    if voxceleb2_root:
+        logging.info("Preparing VoxCeleb2...")
+        v2_manifests = _prepare_voxceleb_v2(voxceleb2_root, num_jobs)
+        if "train" in manifests:
+            manifests["train"]["recordings"] = combine(
+                manifests["train"]["recordings"], v2_manifests["recordings"]
+            )
+            manifests["train"]["supervisions"] = combine(
+                manifests["train"]["supervisions"], v2_manifests["supervisions"]
+            )
+        else:
+            manifests["train"] = v2_manifests
+
+    for split in ("train", "test"):
+        recordings = manifests[split]["recordings"]
+        supervisions = manifests[split]["supervisions"]
+        validate_recordings_and_supervisions(recordings, supervisions)
+        if output_dir is not None:
+            recordings.to_json(output_dir / f"voxceleb_recordings_{split}.jsonl.gz")
```

Comment:
```suggestion
            recordings.to_file(output_dir / f"voxceleb_recordings_{split}.jsonl.gz")
```

It’s best to use to file to avoid errors when changing extensions. See other loc saving manifests in this PR too.

Reply:
Thanks! Fixed now.